packages/components/src/tools-panel/tools-panel/hook.ts

```diff
@@ -92,6 +92,27 @@ export function useToolsPanel(
 		setMenuItems( items );
 	}, [ panelItems ] );
 
+	// Track whether optional controls, if any, are displayed or not.
+	// Default state is that all optional controls, if any, are hidden.
+	const [
+		areAllOptionalControlsHidden,
+		setAreAllOptionalControlsHidden,
+	] = useState( true );
+
+	// Where no optional menu items are active, we display a plus icon
+	// to indicate the presence of further menu items.
+	useEffect( () => {
+		if ( menuItems.optional ) {
+			const optionalMenuItemsArray = Object.entries( menuItems.optional );
+			const newValue =
+				optionalMenuItemsArray.length > 0 &&
+				! optionalMenuItemsArray.some(
+					( [ , isSelected ] ) => isSelected
+				);
+			setAreAllOptionalControlsHidden( newValue );
+		}
+	}, [ menuItems.optional ] );
+
```

Comment:
Unless I'm missing something here the code here could stay as it is but we combine the comments to cover both cases. Perhaps something like:

```suggestion
	// Whether all optional menu items are hidden or not must be tracked
	// in order to later determine if the panel display is empty and handle
	// conditional display of a plus icon to indicate the presence of further 
	// menu items.
```

Reply:
🤦 

That this hook was [committed already 5 days ago](https://github.com/WordPress/gutenberg/commit/27cf9444ac5db0fce6184221a39ddd54dc37d0dc#diff-4f87334152aa083c77befd564ae2e8db3c95353c00d7e731abd2dbe883c5fdb5R112) threw me :D My brain short-circuited and didn't expect it to be in there in the rebase.